osde2e/splunk_forwarder_operator_tests.go

```diff
@@ -0,0 +1,166 @@
+// +build osde2e
+
+package osde2etests
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/osde2e-common/pkg/clients/openshift"
+	"github.com/openshift/osde2e-common/pkg/gomega/assertions"
+	sfv1alpha1 "github.com/openshift/splunk-forwarder-operator/api/v1alpha1"
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+)
+
+var (
+	resourceClient *resources.Resources
+	restConfig     *rest.Config
+	deploymentName = "splunk-forwarder-operator"
+	operatorName   = "splunk-forwarder-operator"
+	serviceNames   = []string{"splunk-forwarder-operator-metrics",
+		"splunk-forwarder-operator-catalog"}
+	clusterRoles = []string{
+		"splunk-forwarder-operator",
+		"splunk-forwarder-operator-og-admin",
+		"splunk-forwarder-operator-og-edit",
+		"splunk-forwarder-operator-og-view",
+	}
+	rolePrefix          = "splunk-forwarder-operator"
+	clusterRoleBindings = []string{
+		"splunk-forwarder-operator-clusterrolebinding",
+	}
+	splunkforwarder_names = []string{
+		"osde2e-dedicated-admin-splunkforwarder-x",
+		"osde2e-splunkforwarder-test-2",
+	}
+	testsplunkforwarder           = "osde2e-splunkforwarder-test-2"
+	dedicatedadminsplunkforwarder = "osde2e-dedicated-admin-splunkforwarder-x"
+	operatorNamespace             = "openshift-splunk-forwarder-operator"
+	operatorLockFile              = "splunk-forwarder-operator-lock"
+)
+
+// Blocking SplunkForwarder Signal
+var _ = ginkgo.Describe("Splunk Forwarder Operator", ginkgo.Ordered, func() {
+
+	ginkgo.BeforeAll(func() {
+		// setup the k8s client
+		restConfig, err := config.GetConfig()
+		Expect(err).Should(BeNil(), "failed to get kubeconfig")
+		resourceClient, err = resources.New(restConfig)
+		Expect(err).Should(BeNil(), "resources.New error")
+		Expect(sfv1alpha1.AddToScheme(resourceClient.GetScheme())).Should(BeNil(), "unable to register sfv1alpha1 api scheme")
+
+	})
+	ginkgo.It("is installed", func(ctx context.Context) {
+
+		ginkgo.By("checking the namespace exists")
+		err := resourceClient.Get(ctx, operatorNamespace, operatorNamespace, &corev1.Namespace{})
+		Expect(err).Should(BeNil(), "namespace %s not found", operatorNamespace)
+
+		ginkgo.By("checking the clusterrolebindings exist")
+		var rolebindings rbacv1.RoleBindingList
+		err = resourceClient.List(ctx, &rolebindings)
+		Expect(err).Should(BeNil(), "failed to list rolebindings")
+		found := false
+		for _, rolebinding := range rolebindings.Items {
+			if strings.HasPrefix(rolebinding.Name, rolePrefix) {
+				found = true
+			}
+		}
+		Expect(found).To(BeTrue(), "unable to find clusterrolebindings with prefix %s", rolePrefix)
```

Comment:
Can we update these (all the rbac list checks below) similar to how I did here? https://github.com/openshift/osd-metrics-exporter/pull/102/files#diff-913d35c4b7351573608915863bb92530eb8cd7aa8108b96868c467fdf71f2881R60-R64 this is one of our custom matchers